assets/app/components/file-explorer/component.js

```diff
@@ -59,21 +67,10 @@ export default Ember.Component.extend({
     this.goToDirectory(dir);
   },
 
-  loadDirectory() {
-    var path = this.pathToString();
-    this.get('store').query('file', { filename: path })
-      .then(function(response) {
-        this.set('items', response);
-      }.bind(this));
-  },
-
   goToDirectory(folder) {
-
-    // removing from current
-    var offset = this.get('history_offset');
-    this.get('history').splice(offset, this.get('history').length - offset);
-
     this.get('history').pushObject(folder);
+    var len = this.get('history').get('length');
+    this.get('history').splice(this.get('history_offset'), (len-1) - this.get('history_offset'));
```

Comment:
`len` is guaranteed to equal 1 or above. It represents the number of subdirectories you are currently in. Shortest path in the directory tree is `C:\` where `len == 1`.